src/platforms/android/androidcontroller.cpp

```diff
@@ -268,9 +268,7 @@ void AndroidController::startActivityForResult(JNIEnv* env, jobject /*thiz*/,
 
     if (resultCode == ACTIVITY_RESULT_OK) {
       logger.debug() << "VPN PROMPT RESULT - Accepted";
-      AndroidUtils::dispatchToMainThread(
-          [&] { s_instance->resume_activate(); });
-
+      s_instance->resume_activate();
```

Comment:
can you add an assert to check we are on the main-thread here?

Reply:
we are not on the main thread here - also resume_activate only calls AndroidVPNActivity::sendToService(...) - which can be called from any thread.